gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -482,18 +483,29 @@ private void initializeDelegationTokenSupport(Configuration config, URI path) th
     dts.bindToFileSystem(this, service);
     try {
       dts.init(config);
+      dts.start();
       delegationTokens = dts;
       if (delegationTokens.isBoundToDT()) {
         logger.atFine().log(
             "initializeDelegationTokenSupport(config: %s, path: %s):"
                 + " using existing delegation token",
             config, path);
       }
-    } catch (IllegalStateException e) {
+    } catch (ServiceStateException | IllegalStateException e) {
       logger.atFiner().withCause(e).log("Failed to initialize delegation token support");
     }
   }
 
+  private void stopDelegationTokens() {
+    if (delegationTokens != null) {
+      try {
+        delegationTokens.close();
+      } catch (IOException e) {
+        logger.atSevere().withCause(e).log("Failed to stop delegation tokens support");
```

Comment:
Can you explain why it is OK to swallow the exception?

Reply:
This code is only invoked when the FileSystem close() method is called (i.e., the FS is no longer needed). Since this ultimately triggers the AbstractService#stop() invocation, the only real possibility of encountering an Exception here is if the underlying binding implements serviceStop() and throws an exception therefrom. In which case, I don't think the Exception should be propagated. It is logged here, but what can be done by the FileSystem or its closer in terms of handling the error? Probably nothing. And while it's possible that some resource(s) might be leaked as a consequence, I believe the log message is sufficient (note that it's level SEVERE).